Sources/MacroTesting/AssertMacro.swift

```diff
@@ -343,6 +349,57 @@ public func assertMacro(
   }
 }
 
+// From: https://github.com/apple/swift-syntax/blob/d647052/Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift
```

Comment:
Instead of inlining these in the library, what do you think of putting it in a file in the `SwiftSyntax` directory alongside `SourceEdit`, etc.?

Reply:
Sure, I'm happy to do that. I didn't do so initially because the upstream extensions are `fileprivate` and kept close to the asserts that use them. Plus, these two extensions are just a small parts of their larger upstream file, and I think files from upstream have so far been pulled in wholesale with minimal alterations.

Here are the upstream extensions' signatures from `SwiftSyntax`'s `SwiftSyntaxMacrosTestSupport/Assertions.swift`:

```swift
extension FixIt.Change {
  fileprivate func edit(
    in expansionContext: BasicMacroExpansionContext
  ) -> SourceEdit {
    …
  }
}
```

```swift
extension BasicMacroExpansionContext {
  fileprivate func position(
    of position: AbsolutePosition,
    anchoredAt node: some SyntaxProtocol
  ) -> AbsolutePosition {
    …
  }
}
```

I think it'd be okay to make the extensions `internal` instead of `fileprivate`. Here are some options:

1. Since `BasicMacroExpansionContext` is common between the two extensions, make the extensions `internal` and add them to a new file: `Sources/SwiftSyntax/SourceEditMacroExpansion.swift`.
2. Make the extensions `internal` and them directly to `Sources/SwiftSyntax/SourceEdit.swift`.
3. If you'd rather leave them in `AssertMacro.swift` as `fileprivate`, I can move them to the end of the file.

For me to move forward, feel free to pick one of these ideas or throw in another that you think might work better. Looking forward to your input!